src/KilnUniV3.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import {KilnBase, GemLike} from "./KilnBase.sol";
+import {TwapProduct}       from "./uniV3/TwapProduct.sol";
+
+// https://github.com/Uniswap/v3-periphery/blob/b06959dd01f5999aa93e1dc530fe573c7bb295f6/contracts/SwapRouter.sol#L132
+interface SwapRouterLike {
+    function exactInput(ExactInputParams calldata params) external returns (uint256 amountOut);
+    function factory() external returns (address factory);
+}
+
+// https://github.com/Uniswap/v3-periphery/blob/b06959dd01f5999aa93e1dc530fe573c7bb295f6/contracts/interfaces/ISwapRouter.sol#L26
+// https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps#input-parameters
+struct ExactInputParams {
+    bytes   path;
+    address recipient;
+    uint256 deadline;
+    uint256 amountIn;
+    uint256 amountOutMinimum;
+}
+
+contract KilnUniV3 is KilnBase, TwapProduct {
+    uint256 public scope; // [Seconds]  Time period for TWAP calculations
+    uint256 public yen;   // [WAD]      Relative part of the TWAP's price to insist on
```

Comment:
`part` or `multiplier`? as actually depending on the value of `yen` we could ask for greater, equal or less than the avg price.

Reply:
changed on [e8c0964](https://github.com/makerdao/dss-kiln/pull/9/commits/e8c0964fe1a80e828163e3aad9a8caaf22e3fc9e)